smartsim/_core/launcher/step/step.py

```diff
@@ -66,10 +67,20 @@ def _create_unique_name(entity_name: str) -> str:
         step_name = entity_name + "-" + get_base_36_repr(time.time_ns())
         return step_name
 
+    @staticmethod
+    def ensure_output_directory_exists(output_dir: str) -> None:
+        if not osp.exists(output_dir):
+            pathlib.Path(output_dir).mkdir(parents=True, exist_ok=True)
+
     def get_output_files(self) -> t.Tuple[str, str]:
```

Comment:
Indicate that this a private function with a `_`

Reply:
done!